docs/data/joy/components/stack/ZeroWidthStack.tsx

```diff
@@ -0,0 +1,57 @@
+import * as React from 'react';
+import Avatar from '@mui/joy/Avatar';
+import Box from '@mui/joy/Box';
+import Sheet from '@mui/joy/Sheet';
+import Stack from '@mui/joy/Stack';
+import { styled } from '@mui/joy/styles';
+import Typography from '@mui/joy/Typography';
+
+const Item = styled(Sheet)(({ theme }) => ({
+  ...theme.typography.body2,
+  padding: theme.spacing(1),
+  textAlign: 'center',
+  color: theme.vars.palette.text.tertiary,
+  maxWidth: '400px',
+}));
+
+const message = `Truncation should be conditionally applicable on this long line of text
+ as this is a much longer line than what the container can support. `;
+
+export default function ZeroWidthStack() {
+  return (
+    <Box sx={{ flexGrow: 1, overflow: 'hidden', px: 3 }}>
+      <Item
+        sx={{
+          my: 1,
+          mx: 'auto',
+          p: 2,
+        }}
+      >
+        <Stack spacing={2} direction="row" alignItems="center">
+          <Stack>
+            <Avatar>W</Avatar>
+          </Stack>
+          <Stack>
+            <Typography noWrap>{message}</Typography>
+          </Stack>
+        </Stack>
```

Comment:
```suggestion
        <Stack spacing={2} direction="row" alignItems="center">
          <Avatar>W</Avatar>
          <Typography noWrap>{message}</Typography>
        </Stack>
```

There is no need to double the Stack